### ISSUE TITLE
Trackpad pinch zoom is too sensitive

### ISSUE THREAD

**Issue by diachedelic** [Leaflet/Leaflet#9377]
### Checklist

- [X] I've looked at the [documentation](https://leafletjs.com/reference.html) to make sure the behavior isn't documented and expected.
- [X] I'm sure this is an issue with Leaflet, not with my app or other dependencies (Angular, Cordova, React, etc.).
- [X] I've searched through the [current issues](https://github.com/Leaflet/Leaflet/issues) to make sure this hasn't been reported yet.
- [X] I agree to follow the [Code of Conduct](https://github.com/Leaflet/Leaflet/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.

### Steps to reproduce

1. Open any Leaflet map on a device with a trackpad, such as a laptop.
2. Zoom the map using a pinch gesture.
3. Without removing your fingers from the trackpad, very slowly move your fingers apart or together.

### Expected behavior

The pinch gesture should change the map's zoom level in a gradual and controlled manner, as it does on a touch device.

### Current behavior

Each tiny movement of the fingers changes the zoom level by at least 1, leading to disorientatingly fast changes to the map's bounds. With almost no movement of the fingers, it is possible to take the map from the minimum to the maximum zoom.

### Minimal example reproducing the issue

https://jsfiddle.net/user2314737/tvwkgo34/

### Environment

- Leaflet version: 1.9.4
- Browser (with version): Chrome v126, Safari v17.5 (not Firefox, which does not seem to support [pinch to zoom](https://kenneth.io/post/detecting-multi-touch-trackpad-gestures-in-javascript) on a Mac)
- OS/Platform (with version): MacOS 14.5

Note that these are only the browsers I have tested with. I believe the problem is present on any device with a trackpad.

**Comment by diachedelic** [comment]
I have fixed this bug by reworking the `ScrollWheelZoom` class. If there is a chance it would be accepted, I can submit a PR.

**Comment by diachedelic** [comment]
Here is my patch if anyone is desperate: https://gist.github.com/diachedelic/633712a4c3fc617e9f5e94bdf33c202b.

**Comment by Falke-Design** [comment]
Hi @diachedelic interesting approach, have you tested your changes with a mouse wheel too? Are there any significant differences?

It would be nice to have an PR, but I can't say if we will merge it but we are always happy to see different approaches for the wheel zooming.

**Comment by IvanSanchez** [comment]
Wait a second here - mac trackpad pinch is emulated as **Ctrl+scrollwheel**?!?!

**Comment by diachedelic** [comment]
Yes. That seems to be [a de facto Web standard](https://developer.mozilla.org/en-US/docs/Web/API/WheelEvent#browser_compatibility), not limited to Apple devices.

I have tested my changes with a real mouse wheel, as well as both scrolling and pinching gestures on a trackpad. I also tested it with `zoomSnap` values of 0, 0.5, and 1. There are a couple of constants that still need calibrating, but it feels good.

Note that the `wheelDebounceTime` option has been replaced with a new `maxWheelZoomSpeed` option. This option provides a way to limit the rate of wheel zooming, 10 zoom levels per second by default. Higher values risk disorienting the user, though it can be set to `Infinity`. For my application, I set it to 3 because my users tend to stay within zoom levels 13 and 19.

**Comment by IvanSanchez** [comment]
On one hand, I'd like to tackle down https://github.com/Leaflet/Leaflet/issues/4610 / https://github.com/Leaflet/Leaflet/issues/4696 (or rather, *have the time and mood* to tackle down those). 

On the other hand, I really like @diachedelic 's code and research. Please make it a PR instead of a gist, 'cause I think it should be merged.

**Comment by diachedelic** [comment]
Sure thing, I'll submit a PR once I've calibrated the constants.

**Comment by jcphill** [comment]
Your fiddle is using Leaflet 1.0.0-beta.2. I don't see the problem they way you describe it with 1.94. If anything the trackpad moves too slowly, by only zoomSnap per gesture.

**Comment by diachedelic** [comment]
Oops, I used an off-the-shelf Fiddle and didn't check the versions. I've updated it.

Using a trackpad, try performing a pinch gesture very, very slowly. Even the most minute movement causes a zoom level change.

**Comment by diachedelic** [comment]
That's the strategy I used in my PR: https://github.com/Leaflet/Leaflet/pull/9378/files#diff-440fb039d6d956f84056c30002d6229ee8e55d09542bebfa1f8d0a75e1f70fd3R283.

**Comment by JFWdata** [comment]
See precise work-around by patching :

Asset : settingStore-CwNB8aKw.js
In folder : \ComfyUI_windows_portable\python_embeded\Lib\site-packages\comfyui_frontend_package\static\assets

https://github.com/Comfy-Org/ComfyUI_frontend/issues/17729
